.github/workflows/load-test.yaml

```diff
@@ -0,0 +1,139 @@
+name: On-demand load test
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        required: true
+        description: SHA or branch name to deploy
+        type: string
+        default: 'main'
+      N:
+        required: true
+        description: large N will generate more tuples
+        type: number
+        default: 1
+      M:
+        required: true
+        description: large M implies a single user has a relationship with multiple objects
+        type: number
+        default: 1
+      K:
+        required: true
+        description: large K implies a single object has a relationship with multiple users
+        type: number
+        default: 1
+      testCase:
+        required: true
+        description: the test case to run (see https://github.com/openfga/benchmarks)
+        type: choice
+        options:
+          - 1
+          - 2
+          - 3
+          - 4
+          - 5a
+          - 5b
+          - 6
+          - 7
+          - 8
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: AWSLoadTest
+    # needed for aws-actions/configure-aws-credentials. See https://github.com/aws-actions/configure-aws-credentials/issues/271
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+        with:
+          ref: ${{ github.event.inputs.ref }}
+
+      - name: Set variables
+        id: vars
+        run: echo "sha_short=$(git rev-parse --short=8 ${{ github.event.inputs.ref }})" >> $GITHUB_OUTPUT
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@4b4e9c3e2d4531116a6f8ba8e71fc6e2cb6e6c8c # v2.5.0
+
+      - name: Cache Docker layers
+        uses: actions/cache@88522ab9f39a2ea568f7027eddc7d8d8bc9d59c8 # v3.3.1
+        with:
+          path: /tmp/.buildx-cache
+          key: ${{ runner.os }}-buildx-${{ github.event.inputs.ref }}
```

Comment:
This would only cache for a single commit - maybe we should cache across commits with a deterministic key?

Something like: `${{ runner.os }}-buildx-1` that we can always change if we need to evict cache

Reply:
Removed this step